src/collection/backend/lmdb.cc

```diff
@@ -498,22 +499,28 @@ void LMDB::resolveMultiMatches(const std::string& var,
         while ((rc = mdb_cursor_get(cursor, &key, &data, MDB_NEXT)) == 0) {
             l->insert(l->begin(), new VariableValue(
                 &m_name,
-                new std::string(reinterpret_cast<char *>(key.mv_data),
+                std::string(reinterpret_cast<char *>(key.mv_data),
                 key.mv_size),
-                new std::string(reinterpret_cast<char *>(data.mv_data),
+                std::string(reinterpret_cast<char *>(data.mv_data),
                 data.mv_size)));
         }
     } else {
-        while ((rc = mdb_cursor_get(cursor, &key, &data, MDB_NEXT)) == 0) {
-            char *a = reinterpret_cast<char *>(key.mv_data);
-            if (strncmp(var.c_str(), a, keySize) == 0) {
-                l->insert(l->begin(), new VariableValue(
-                    &m_name,
-                    new std::string(reinterpret_cast<char *>(key.mv_data),
-                    key.mv_size),
-                    new std::string(reinterpret_cast<char *>(data.mv_data),
-                    data.mv_size)));
-            }
+        string2val(var, &key);
+        rc = mdb_cursor_get(cursor, &key, &data, MDB_SET_RANGE);
```

Comment:
@ziollek replace `MDB_SET_RANGE ` to `MDB_SET_KEY`
`MDB_SET_RANGE` - getting first key greater than or equal to specified key [http://www.lmdb.tech/doc/group__mdb.html#ga1206b2af8b95e7f6b0ef6b28708c9127](url)
If you are looking for none exist `samplekey` you get key containing `samplekey` like `samplekey2` (if key exist)

Reply:
Workable link for @sobczak-m comment's - http://www.lmdb.tech/doc/group__mdb.html#gga1206b2af8b95e7f6b0ef6b28708c9127af9feb0557c2954dbf7732eee5e1b59e7